sql/hive/src/main/scala/org/apache/spark/sql/hive/orc/OrcFileFormat.scala

```diff
@@ -54,10 +57,12 @@ class OrcFileFormat extends FileFormat with DataSourceRegister with Serializable
       sparkSession: SparkSession,
       options: Map[String, String],
       files: Seq[FileStatus]): Option[StructType] = {
-    OrcFileOperator.readSchema(
-      files.map(_.getPath.toUri.toString),
-      Some(sparkSession.sessionState.newHadoopConf())
-    )
+    // Safe to ignore FileNotFoundException in case no files are found.
+    val schema = Try(OrcFileOperator.readSchema(
```

Comment:
@rajeshbalamohan is this change unnecessary for this PR? If so, I'd like to revert it to make the PR as small as possible.